californium-core/src/main/java/org/eclipse/californium/core/network/stack/ReliabilityLayer.java

```diff
@@ -50,15 +49,16 @@
 
 	/** The random numbers generator for the back-off timer */
 	private Random rand = new Random();
-	
+
 	private int ack_timeout;
 	private float ack_random_factor;
 	private float ack_timeout_scale;
 	private int max_retransmit;
 
 	/**
-	 * Constructs a new reliability layer.
-	 * Changes to the configuration are observed and automatically applied.
+	 * Constructs a new reliability layer. Changes to the configuration are
+	 * observed and automatically applied.
```

Comment:
On a first view, it's simply not working!
 
        NetworkConfig 
	public NetworkConfig set(String key, Object value) {
		if (key == null) {
			throw new NullPointerException("key must not be null");
		} else if (value == null) {
			throw new NullPointerException("value must not be null");
		} else {
			properties.put(key, String.valueOf(value));
			for (NetworkConfigObserver obs:observers) {
				obs.changed(key, value);
			}
			return this;
		}
	}

only calls the `changed(... Object)` method. 
The `NetworkConfigObserverAdapter `doesn't convert to calls of  the others. So `int`  and `float` are never called.

if they would have been called, I have no idea, how this was intended to be "thread safe".

So I think, remove it completly (but in an other PR).